flips/20211007-transaction-fees.md

```diff
@@ -0,0 +1,264 @@
+
+This separation into a part that can be know before the transaction is executed, and a part that cannot be know before the transaction is executed, allows for better oversight from the user into how much the transaction will actually cost.
+
+Each part of the fees can be further split into *effort*(<img src="https://render.githubusercontent.com/render/math?math=E">) and an *effort cost factor*(<img src="https://render.githubusercontent.com/render/math?math=f">) of FLOW per unit of effort. The *effort* required for a specific transaction should be independent of when the transaction is sent[^1] and should only depend on the properties of the transaction. On the other hand the *effort cost factor* would change over time depending on network activity.
+
+[^1]: Assuming there weren't any changes to the state that would make the transactions script execute differently.
+
+The final result in equation form looks like this:
+
+<img src="https://render.githubusercontent.com/render/math?math=F%5E%7Bmax%7D%20%3D%20f%20(%20E_i%20%2B%20E_e%20)">
+
+<!-- TODO update diagram -->
+
+![Alt text](20211007-transaction-fees/fees_diagram.svg)
+
+- <img src="https://render.githubusercontent.com/render/math?math=F_i">: Inclusion fees; Represent the cost of including the transaction in a collection.
+    - <img src="https://render.githubusercontent.com/render/math?math=f">: The current cost of one unit of effort.
+    - <img src="https://render.githubusercontent.com/render/math?math=E_i">: Inclusion effort; The time invariant effort required to handle this transaction (include it into a collection, check signatures are valid, check that the payer can pay, transfer the transaction over the wire,...). The inclusion effort is known before the transaction is executed.
+    - <img src="https://render.githubusercontent.com/render/math?math=E_c">: Execution effort; The time invariant effort required to execute this transaction (run the script, process the results, send results to verification node, verify the results, ...). The Execution effort cannot be know without executing the transactions script.
+
+
+### Fee effort cost factor
+
+The effort cost factor <img src="https://render.githubusercontent.com/render/math?math=f"> should be defined on a smart contract and adjustable via the service account admin resource and should be accessible for everyone to read. 
+
+The effort cost factor can be used in three different ways:
+
+1. The main usage of the effort cost factors is adjusting it when the network is under load. 
```

Comment:
In consideration of surge pricing, I think we should separate traffic as a unique parameter to charge for